### PR TITLE
Update Fennel keywords to catch up to version 0.6.0.

### PR DESCRIPTION
**Pull request by technomancy**
The keywords in the current Fennel lexer are a bit behind; this catches it up to the latest release.

Also removes some disallowed identifier characters we accidentally left in previously.

### CODE REVIEW

**Review by birkenfeld — APPROVED** [review]
JFYI, the `u` prefix can be omitted now that we're py3-only.

**Comment by birkenfeld** [comment]
However, the example file doesn't seem to like your changes?  Can you verify that https://github.com/pygments/pygments/blob/master/tests/examplefiles/fennelview.fnl is still valid?

**Comment by technomancy** [comment]
Thanks! Sorry for the noise; I think this should take care of it now. I had forgotten the previous patch was from such an old version, so the example needed to be brought up to date.

File: pygments/lexers/lisp.py [hunk 1]
```diff
@@ -2670,7 +2675,6 @@ class FennelLexer(RegexLexer):
             (r'-?\d+', Number.Integer),
 
             (r'"(\\\\|\\"|[^"])*"', String),
```
Comment (birkenfeld) [inline pygments/lexers/lisp.py:2677]: Sorry, I just saw this right now: this regex is prone to catastrophic backtracking, we should fix it while at it.  It should look like 
```
"(\\\\|\\"|[^"\\])*"
```
with alterations as needed for other allowed escape sequences.  If they are complex, a separate state can be used as well.

**Review by birkenfeld — APPROVED** [review]
The changes themselves look good now, we just should resolve the string regex issue.

**Review by birkenfeld — APPROVED** [review]

**Comment by birkenfeld** [comment]
I made another small fix to account for other allowed escapes.

**Comment by technomancy** [comment]
Thanks! As you can probably tell, I've never written Python before so I appreciate the cleanup.

**Comment by birkenfeld** [comment]
Thanks even more for the contribution then!

File: pygments/lexers/lisp.py (hunk 1 above)
Comment (Anteru) [reply pygments/lexers/lisp.py]: That's literally the third time in a week or so we got the same regex of doom :)

Comment (technomancy) [reply pygments/lexers/lisp.py]: Oh, well really that regex was added years ago; copied from an existing lexer IIRC. But I'm glad to see it gone.

**Pull request merged**